Paperspace/gradient-cli
# CR2-22 CR2-49 CR2-48 add metrics list to sdk

Adding list custom metrics functionality to sdk and cli for jobs, experiments, deployments, and notebooks

QA Test Plan:
1. in cli, check that this new `list` command shows up in the help text: `gradient deployments metrics --help`
2. create a deployment (gui or cli doesn't matter) and wait for it to finish provisioning
3. try the new `list` command: `gradient deployments metrics list --id XYZ`, should return custom metrics (a list of words),
4. repeat 1 and 2 for jobs, experiments, and notebooks. these might return `null` but that's okay for now, just checking that this new command is available. note the stuff that returns `null` below if any and I'll solve that in a separate ticket.
